src/aatoolbox/datasources/chirps/chirps.py

```diff
@@ -0,0 +1,631 @@
+            date_list = [
+                (d.split("-")[0], d.split("-")[1], d.split("-")[2])
+                for d in pd.date_range(
+                    f"{start_year}-{start_month}-{start_day}",
+                    f"{end_year}-{end_month}-{end_day}",
+                    freq="D",
+                )
+                .strftime("%Y-%m-%d")
+                .tolist()
+            ]
+
+        # Create a sentence containing information on the downloaded data
+        if len(date_list) == 0:
+            sentence_to_print = (
+                "No data will be downloaded. "
+                "There is no {self._frequency} available within the "
+                "chosen range of dates."
+            )
+        else:
+            sentence_to_print = (
+                f"{self._frequency.capitalize()} "
+                "data will be downloaded, starting from "
+                f"{'-'.join(date_list[0])} to {'-'.join(date_list[-1])}."
+            )
+
+        return date_list, sentence_to_print
+
+    def _get_file_name(
+        self,
+        year: Optional[Union[int, str]],
+        month: Optional[Union[int, str]],
+        day: Optional[Union[int, str]],
+    ) -> str:
+        # Set wildcard for year, to get general filename pattern
+        # used to find all files with pattern in folder
+        if year is None:
+            year = "*"
+        if month is None:
+            month = "*"
+        else:
+            if len(f"{month}") == 1:
+                month = f"0{month}"
+        if day is None:
+            day = "*"
+        else:
+            if len(f"{day}") == 1:
+                day = f"0{day}"
+
+        file_name_base = (
+            f"{self._country_config.iso3}_chirps_"
+            f"{self._frequency}_{year}_{month}_"
+        )
+        if self._frequency == "daily":
+            file_name_base += f"{day}_"
+        file_name = (
+            f"{file_name_base}"
+            f"r{self._resolution}"
+            f"_{self._geobb.get_filename_repr(p=0)}.nc"
+        )
+        return file_name
+
+    def _get_raw_path(
+        self, year: Optional[str], month: Optional[str], day: Optional[str]
+    ) -> Path:
+        return self._raw_base_dir / self._get_file_name(
+            year=year, month=month, day=day
+        )
+
+    def _get_processed_path(self, raw_path: Path) -> Path:
+        return self._processed_base_dir / raw_path.parts[-1]
+
+    def _get_url(self, year: str, month: str, day: str) -> str:
+
+        # Convert month from month number (in string format) to
+        # three-letter name
+        month = calendar.month_abbr[int(month)]
+
+        if self._frequency == "daily":
+            url = (
+                "https://iridl.ldeo.columbia.edu/SOURCES/.UCSB/"
+                f".CHIRPS/.v2p0/.daily-improved/.global/."
+                f"{str(self._resolution).replace('.', 'p')}/.prcp/"
+                f"T/%28{day}%20{month}%20{year}%29%28{day}%20{month}%20{year}"
+                "%29RANGEEDGES/data.nc"
```

Comment:
!!! You are not including the geobounding coordinates here, i.e. you are always downloading the whole globe!! 
I discovered this by running the example in the documentation you wrote, so would be good to also run this example yourself and check if it is giving the desired result.